uber/templates/registration/promo.html

```diff
@@ -0,0 +1,48 @@
+{% extends "base-admin.html" %}
+{% block title %}Generate a PromoCode{% endblock %}
+{% block content %}
+
+<h2> Create a Promotional Badge Code</h2>
+
+<form method="post" action="promo">
+{% csrf_token %}
+<table class="form">
+<tr>
+    <td> <b>Expiration Date:</b> </td>
+    <td>
+        <input type="date" name="expiration_date" id="datefield"/>
```

Comment:
Based on what I've seen, it seems you don't intend for admins to be able to edit promo codes. This is fine; however, you will still need to write this template as if you intended that. That means:
1. Passing the `promo_code` object to the template (via `return`), and
2. Making each field's value equal to the object's value, e.g., `promo_code.expiration_date`

That's because, when you trip an error and return the form, the page reloads. If you DON'T do these things, the page will reload with a blank form. That will be super super frustrating.

As stated in a previous comment, once you do this you'll want to make sure you only `return` to the page if the admin didn't fill out the form correctly. Otherwise, when they succeed in making a code, they'll get a form with all the fields filled out already, which you probably don't want.